### DISCUSSION TITLE
glog doesn't work with non-CMake

### DISCUSSION THREAD

**Discussion by OrientalSiru** [microsoft/vcpkg#39300] [Q&A] [answered]
I am using the non-CMake version of Visual Studio, and I installed glog with vcpkg. However, I get the following error when trying to include the library:

`Error  C1189   #error:  <glog/logging.h> was not included correctly. See the documention for how to consume the library.`

Tried to resolve the above by adding #define GLOG_USE_GLOG_EXPORT, I get another error:

`Error  C4996  google::CustomPrefixCallback': Use PrefixFormatterCallback instead.`

Opened up a [question](https://github.com/google/glog/issues/1107) on glog github, and they confirmed that they only support Bazel & CMake. Has anyone managed to get the non-CMake to work for glog? Is the expectation that every vcpkg package should work with non-CMake? Could the doc be updated if the expecation is incorrect (currently it sounds like it should just work magically on Visual Studio)?

**Comment by AugP** [comment, marked as answer]
By "non-CMake version of Visual Studio", do you mean MSBuild? In general, it is up to library authors to determine their level of integration with different build systems, operating systems, or architectures. vcpkg does not make any guarantees of support beyond what the library already offers. vcpkg broadly integrates with MSBuild, but not every individual library may work properly with MSBuild. If the library authors explicitly only support Bazel and CMake, it sounds like you would need to use one of those build systems with that library unless someone submits a PR to them to add support for MSBuild.

**Comment by annerose** [comment]
I encountered the same issue while using vcpkg glog 0.7.1 in VS2019. I believe it's a bug in glog that failed to handle MSBuild's 4996. Here's how I fixed it:

``` c++
#if defined(_MSC_VER)
#  pragma warning(push)
#  pragma warning(disable : 4996)
#endif  // _MSC_VER

#include <glog/logging.h>

#if defined(_MSC_VER)
#  pragma warning(pop)
#endif  // _MSC_VER

**Reply by Osyotr** [reply]
It's neither vcpkg nor msvc/msbuild issue.
Your project uses deprecated functionality of glog together with some setting that makes this an error (probably `/sdl`). As such, you should fix your project instead.
https://github.com/google/glog/blob/7b134a5c82c0c0b5698bb6bf7a835b230c5638e4/src/glog/logging.h#L495
